File: Euler-Particles/src/color/HSB/RainbowColorGenerator.java
package color.HSB;

import processing.core.PApplet;

public class RainbowColorGenerator extends HSBColorGenerator {

    public RainbowColorGenerator(PApplet p) {
        super(p);
    }

    public int mapColor(float value, float minValue, float maxValue, int saturation, int brightness) {
        p.colorMode(p.HSB, 360);
        int hue = (int) PApplet.map(value, minValue, maxValue, 0, 360);
        return p.color(hue, saturation, brightness);
    }

}

File: docs/Collisions/color/hsb/rainbow_color_generator.js
class RainbowColorGenerator extends HSBColorGenerator {

    mapColor(value, minValue, maxValue, saturation, brightness) {
        colorMode(HSB, 360);
        let hue = int(map(value, minValue, maxValue, 0, 360));
        return color(hue, saturation, brightness);
    }
}
